### DISCUSSION TITLE
initialization of ScenegraphLayer - Assertion failed

### DISCUSSION THREAD

**Discussion by sunyonglincanada** [visgl/deck.gl#5276] [Q&A] [answered]
Hi, 

I‘ve got one question that I am trying to load ScenegraphLayer using the object from the cache by LocalForage. But it pops up an error - deck: error during initialization of ScenegraphLayer({id: 'ScenegraphLayer'}) Error: luma.gl: assertion failed.
at assert (0.chunk.js:56955) The thing is I logged the object from the cache before sending it in the scenegraph and it is the same as the one that using gltf loader - https://imgur.com/a/H1SzTLQ

I was using the ScenegraphLayer along with GLTF loader to load content from the glb file. The logic is I want to save the glb file using the loader to cache when the file loads the first time, then I will fetch the object from cache and assign it to the ScenegraphLayer as the value of scenegraph. 

Here is the gist for my code - https://gist.github.com/sunyonglincanada/6fe79b8035b8c3f045abd4f37d1e0ca0

Can any of you help me out? Thanks in advance!!

**Comment by Pessimistress** [comment, marked as answer]
You cannot serialize and deserialize typed arrays.

```
> var a = new Uint8Array([0, 1, 2, 3])
> a
Uint8Array(4) [0, 1, 2, 3]
> a = JSON.stringify(a)
"{"0":0,"1":1,"2":2,"3":3}"
> JSON.parse(a)
{0: 0, 1: 1, 2: 2, 3: 3}
```

**Reply by sunyonglincanada** [reply]
Ah, you are right, I've got the way to send the scenegraph data to cache finally lol. Thanks so much for the reminder and help!! 😁
